### PR TITLE
specload: don't allow NULL as parameter in elektraSpecloadSendSpec

### PR DESCRIPTION
**Pull request by hannes99**
if parentKey was NULL when calling elektraSpecloadSendSpec dumping the spec failed because the quickdump parent key was not set correctly.

## Basics

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [x] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [x] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

File: doc/news/_preparation_next_release.md [hunk 1]
```diff
@@ -67,11 +67,9 @@ The following section lists news about the [plugins](https://www.libelektra.org/
 - <<TODO>>
 - <<TODO>>
 
-### <<Plugin>>
+### specload
 
-- <<TODO>>
-- <<TODO>>
-- <<TODO>>
+- small fix to always dump to stdout
```
Comment (kodebach) [inline doc/news/_preparation_next_release.md:72]: A bit more explanation would be nice here

**Review by kodebach — CHANGES_REQUESTED** [review]
What is the use case for this change? Why would you call sendspec without a parent key?

IMO there should just be NULL checks for `spec` and `parentKey` at the start and if something is NULL we just return `ELEKTRA_PLUGIN_STATUS_ERROR` immediately.

In any case, you should also add at least one test case for the new behaviour to `testmod_specload`.

**Comment by hannes99** [comment]
https://github.com/ElektraInitiative/libelektra/blob/e3c31c738d3666878635741f2c97e979503f2f23/src/plugins/quickdump/quickdump.c#L488-L492

file can be `stdout`, I'm pretty sure we should not close that(if it is). Am I missing something? @kodebach

**Comment by kodebach** [comment]
Hm, yes just closing `stdout` is a bit questionable. But since it only happens if a write to `stdout` failed it's probably not taht big a deal. If you really want to fix it, feel free to do so...

**Comment by hannes99** [comment]
the context was a test after which `stdout` disappeared, I'll just test failing return values. I guess that checking the actual dump is more of a `quickdumper`, than a `specload`, thing.

**Comment by kodebach** [comment]
> I guess that checking the actual dump is more of a `quickdump`, than a `specload`, thing.

Yes, the `specload` tests definitely shouldn't check the output from `quickdump` beyond a very cursory check that something was produced.

> `file` is always closed at the end, not just in a failing case.

Ah okay, in the success case we shouldn't close `stdout`. That should be fixed.

**Review by kodebach — APPROVED** [review]
Looks good. Great work!

**Pull request merged**